Unify command line arguments terminology

jan-cerny opened the pull request:
#### Description:
Changes `--resolved-rules` in `combine_remediations.py` and `--resolve-rules-into` in `yaml_to_shorthand.py` both to `--resolved-rules-dir`, because they are both describing the same directory. 

Also changes `--output_dir` in `combine_remediations.py` to `--output-dir` to use dashes consistently in the CLI interface.


Also updates the invocations of these scripts in CMake accordingly.


#### Rationale:
Resolves inconsistency created by  #3827 in whichthe directory with resolved rules is referred to as `--resolved-rules`, which differs from the `yaml_to_shorthand` script, which uses more accurate `--resolved-rules-dir`.
Fixes #4027

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/ComplianceAsCode/content/inspections/43f7b7b0-c7c3-4b7b-b764-c489ce950960) was created.

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/ComplianceAsCode/content/inspections/43f7b7b0-c7c3-4b7b-b764-c489ce950960) completed: **No new issues**

yuumasato commented on build-scripts/combine_remediations.py:
```diff
@@ -29,14 +29,14 @@ def parse_args():
+        "--output-dir", required=True, dest="output_dir",
         help="output directory where all remediations will be saved"
     )
     p.add_argument("fixdirs", metavar="FIX_DIR", nargs="+",
```
Please, change this to `fix_dirs` as well.

matejak commented:
This PR needs rebase, I would also remove the `dest=...`, because those are redundant.

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/ComplianceAsCode/content/inspections/e11df3e8-5012-4bd3-a02f-860600153884) was created.

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/ComplianceAsCode/content/inspections/e11df3e8-5012-4bd3-a02f-860600153884) completed: **No new issues**

jan-cerny commented:
I have resolved the conflicts, rebased on top of master, removed redundant `dest` arguments and renamed `fixdirs` variable to `fix_dirs`. I have also replaced underscores by dashes in all options in both scripts.

yuumasato commented:
Great, thank you!
